core/src/main/java/org/apache/iceberg/ClientPool.java

```diff
@@ -26,4 +28,16 @@
   <R> R run(Action<R, C, E> action) throws E, InterruptedException;
 
   <R> R run(Action<R, C, E> action, boolean retry) throws E, InterruptedException;
+
+  /**
+   * Initialize the client pool with catalog properties and an optional hadoop configuration.
+   *
+   * <p>A custom ClientPool implementation must have a no-arg constructor. A Catalog using the
+   * ClientPool will first use this constructor to create an instance of the pool, and then call
+   * this method to initialize the pool.
+   *
+   * @param properties catalog properties
+   * @param hadoopConf hadoop configuration
+   */
+  default void initialize(Map<String, String> properties, Object hadoopConf) {}
```

Comment:
Actually did you consider following FileIO way and only passing it in, if the impl implements Configurable?  Maybe we can avoid the ugly Object here.